testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/microprofile/metrics/MicroProfileMetricsWSTestCase.java

```diff
@@ -0,0 +1,179 @@
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test WS with MP metrics
+ *
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2018 Red Hat inc.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ MicroProfileMetricsWSTestCase.EnablesUndertowStatistics.class })
+public class MicroProfileMetricsWSTestCase {
+
+    static class EnablesUndertowStatistics implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            managementClient.getControllerClient().execute(enableStatistics(true));
+            reloadIfRequired(managementClient);
+        }
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+            managementClient.getControllerClient().execute(enableStatistics(false));
+            reloadIfRequired(managementClient);
+        }
+
+        private ModelNode enableStatistics(boolean enabled) {
+            final ModelNode address = Operations.createAddress(SUBSYSTEM, "undertow");
+            return Operations.createWriteAttributeOperation(address, STATISTICS_ENABLED, enabled);
+        }
+    }
+
+    @Deployment(name = "jaxws-basic-pojo", managed = false)
+    public static Archive<?> deployment() {
+        WebArchive pojoWar = ShrinkWrap.create(WebArchive.class, "jaxws-basic-pojo.war").addClasses(EndpointIface.class,
+                PojoEndpoint.class, HelloObject.class);
+        pojoWar.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return pojoWar;
+    }
+
+    @BeforeClass
+    public static void skipSecurityManager() {
+        AssumeTestGroupUtil.assumeSecurityManagerDisabled();
+    }
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    @ArquillianResource
+    private Deployer deployer;
+
+
+    @Test
+    @InSequence(1)
+    public void testApplicationMetricBeforeDeployment() throws Exception {
+        getPrometheusMetrics(managementClient, "application", false);
+        getJSONMetrics(managementClient, "application", false);
+
+        // deploy the archive
+        deployer.deploy("jaxws-basic-pojo");
+    }
+
+    @Test
+    @InSequence(2)
+    @OperateOnDeployment("jaxws-basic-pojo")
+    public void testDeploymenTest(@ArquillianResource URL url) throws Exception {
+        QName serviceName = new QName("http://jbossws.org/basic", "POJOService");
+        URL wsdlURL = new URL(url, "POJOService?wsdl");
+        Service service = Service.create(wsdlURL, serviceName);
+        EndpointIface proxy = service.getPort(EndpointIface.class);
+        checkRequestCount(2, true);
```

Comment:
out of curiosity: where are these 2 HTTP requests coming from?

Reply:
Theses are two requests to access wsdl and it's imported resource.